### DISCUSSION TITLE
How to access dag_run context in @task.docker decorated task?

### DISCUSSION THREAD

**Discussion by juhi24** [apache/airflow#43158] [Q&A]
```python
@task.docker(image=myimage)
def job(**kws):
    print(kws)
```

The above code would print `{}`. Is the `dag_run` and `task_instance` context not available in the docker decorated tasks? Any workarounds besides using `DockerOperator`?

**Comment by potiuk** [comment]
https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/taskflow.html#context

**Reply by juhi24** [reply]
Yes, this is how I succesfully do it with `@task`. But should it also work with `@task.docker`? When I'm trying to pass `dag_run` as a kw argument to my `@task.docker` task, it comes in as `None`. Seems as if no context is passed to the task.

**Reply by potiuk** [reply]
task in docker is executed in ... docker. Context is not passed to inside docker running in different containers - because it would have to be serialized and you'd have to have airflow installed in your container

**Reply by juhi24** [reply]
I see, thanks. Suppose I had airflow installed inside the container, would the context be passed in that case? Or would it never be passed to a docker task in any case?

**Reply by potiuk** [reply]
Not sure -  look if DockerOperator has this feature (read the docs) - if not, then it would be a nice feature to add, and you could either (best) PR and contribute it, or open an issue describing it.

**Reply by juhi24** [reply]
Thanks for the reply. My reason of writing here is that the docs seemed to be lacking this info. However, the result I described in the original post suggests that no context whatsoever was available in the docker task. With your input, I start believing this is not a bug, but rather the current state of features. I hope to be able to contribute somehow in the coming weeks. Thanks.
